server/website/website/views.py

```diff
@@ -491,20 +489,20 @@ def handle_result_files(session, files, execution_times=None):
     dbms_id = session.dbms.pk
     udm_before = {}
     udm_after = {}
-    if ('user_defined_metrics' not in files) and (USER_DEFINED_TARGET == session.target_objective):
-        return HttpResponse('ERROR: user defined target objective is not uploaded!')
-    # User defined metrics
-    udm = {}
+    udm_all = {}
     if 'user_defined_metrics' in files:
-        udm = JSONUtil.loads(files['user_defined_metrics'])
-    if len(udm) > 0:
-        udm_target = udm['target_objective']
-        udm_not_target = udm['metrics']
-        udm_all = copy.deepcopy(udm_not_target)
-        if (udm_target is None) and (USER_DEFINED_TARGET == session.target_objective):
-            return HttpResponse('ERROR: user defined target objective is not uploaded!')
-        if udm_target is not None:
-            udm_all.update(udm_target)
+        udm_all = JSONUtil.loads(files['user_defined_metrics'])
+    target_name = session.target_objective
+    target_instance = target_objectives.get_instance(dbms_id, target_name)
+    if target_instance.is_udf() and len(udm_all) == 0:
+        return HttpResponse('ERROR: user defined target objective {} is not uploaded!'.format(
+            target_name))
```

Comment:
This part is the same as the below part. You might want to change the message to 'No user_defined_metrics uploaded'.